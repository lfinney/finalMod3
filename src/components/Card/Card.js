import React from 'react';
import PropTypes from 'prop-types';

const Card = ({ house }) => {

  console.log(house);
  const seats = house.seats.map( seat =>
    <p key={seat} >{seat}</p>);
  const titles = house.titles.map( title =>
    <p key={title} >{title}</p>);
  const weapons = house.ancestralWeapons.map( weapon =>
    <p key={weapon} >{weapon}</p>);

  return (
    <div className="card">
      <h2>{house.name}</h2>
      <p>{house.founded}</p>
      <h3>House Seats</h3>
      <h3>{seats}</h3>
      <h3>House Titles</h3>
      <div>{titles}</div>
      <h3>Coat of Arms</h3>
      <p>{house.coatOfArms}</p>
      <h3>Ancestral Weapons</h3>
      <div>{weapons}</div>
      <h3>{house.words}</h3>
    </div>
  );
};

Card.propTypes = {
  house: PropTypes.object
};

export default Card;
