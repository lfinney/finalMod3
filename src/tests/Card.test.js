import React from 'react';
import Card from '../components/Card/Card';
import { shallow, mount } from 'enzyme';
import housesMock from './mockData/housesMock';

describe('Card', () => {

  it('should render component', () => {
    const wrapper = shallow(
      <Card house={housesMock[0]} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render component', () => {
    const wrapper = mount(
      <Card house={housesMock[0]} />);

    const clickableDiv = wrapper.find('.card');

    expect(wrapper.state().showMembers).toEqual(false);
    clickableDiv.simulate('click');
    expect(wrapper.state().showMembers).toEqual(true);
  });
});
