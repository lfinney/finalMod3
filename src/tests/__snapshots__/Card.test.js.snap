// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    house={
        Object {
            "ancestralWeapons": Array [
              "Lady Forlorn",
            ],
            "coatOfArms": "Three black ravens in flight...",
            "diedOut": "",
            "founded": "",
            "name": "House Corbray of Heart's Home",
            "seats": Array [
              "Heart's Home",
            ],
            "swornMembers": Array [
              "Jon Snow",
            ],
            "titles": Array [
              "Lord of Heart's Home",
              "King of the Fingers (historical)",
            ],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          House Corbray of Heart's Home
</h2>,
        <p />,
        <h3>
          House Seats: 
          <span>
                    Heart's Home
          </span>
</h3>,
        <h3>
          House Titles
</h3>,
        <div
          className="flex-data"
>
          <p
                    className="flex-me"
          >
                    Lord of Heart's Home
          </p>
          <p
                    className="flex-me"
          >
                    King of the Fingers (historical)
          </p>
</div>,
        <h3>
          Coat of Arms
</h3>,
        <p>
          Three black ravens in flight...
</p>,
        <h3>
          Ancestral Weapons
</h3>,
        <div>
          <p>
                    Lady Forlorn
          </p>
</div>,
        <h3 />,
        false,
      ],
      "className": "card",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "House Corbray of Heart's Home",
        },
        "ref": null,
        "rendered": "House Corbray of Heart's Home",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "House Seats: ",
            Array [
              <span>
                Heart's Home
</span>,
            ],
          ],
        },
        "ref": null,
        "rendered": Array [
          "House Seats: ",
          Object {
            "instance": null,
            "key": "Heart's Home",
            "nodeType": "host",
            "props": Object {
              "children": "Heart's Home",
            },
            "ref": null,
            "rendered": "Heart's Home",
            "type": "span",
          },
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "House Titles",
        },
        "ref": null,
        "rendered": "House Titles",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="flex-me"
>
              Lord of Heart's Home
</p>,
            <p
              className="flex-me"
>
              King of the Fingers (historical)
</p>,
          ],
          "className": "flex-data",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "Lord of Heart's Home",
            "nodeType": "host",
            "props": Object {
              "children": "Lord of Heart's Home",
              "className": "flex-me",
            },
            "ref": null,
            "rendered": "Lord of Heart's Home",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": "King of the Fingers (historical)",
            "nodeType": "host",
            "props": Object {
              "children": "King of the Fingers (historical)",
              "className": "flex-me",
            },
            "ref": null,
            "rendered": "King of the Fingers (historical)",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Coat of Arms",
        },
        "ref": null,
        "rendered": "Coat of Arms",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Three black ravens in flight...",
        },
        "ref": null,
        "rendered": "Three black ravens in flight...",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Ancestral Weapons",
        },
        "ref": null,
        "rendered": "Ancestral Weapons",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              Lady Forlorn
</p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "Lady Forlorn",
            "nodeType": "host",
            "props": Object {
              "children": "Lady Forlorn",
            },
            "ref": null,
            "rendered": "Lady Forlorn",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h3",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            House Corbray of Heart's Home
</h2>,
          <p />,
          <h3>
            House Seats: 
            <span>
                        Heart's Home
            </span>
</h3>,
          <h3>
            House Titles
</h3>,
          <div
            className="flex-data"
>
            <p
                        className="flex-me"
            >
                        Lord of Heart's Home
            </p>
            <p
                        className="flex-me"
            >
                        King of the Fingers (historical)
            </p>
</div>,
          <h3>
            Coat of Arms
</h3>,
          <p>
            Three black ravens in flight...
</p>,
          <h3>
            Ancestral Weapons
</h3>,
          <div>
            <p>
                        Lady Forlorn
            </p>
</div>,
          <h3 />,
          false,
        ],
        "className": "card",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "House Corbray of Heart's Home",
          },
          "ref": null,
          "rendered": "House Corbray of Heart's Home",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "House Seats: ",
              Array [
                <span>
                  Heart's Home
</span>,
              ],
            ],
          },
          "ref": null,
          "rendered": Array [
            "House Seats: ",
            Object {
              "instance": null,
              "key": "Heart's Home",
              "nodeType": "host",
              "props": Object {
                "children": "Heart's Home",
              },
              "ref": null,
              "rendered": "Heart's Home",
              "type": "span",
            },
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "House Titles",
          },
          "ref": null,
          "rendered": "House Titles",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="flex-me"
>
                Lord of Heart's Home
</p>,
              <p
                className="flex-me"
>
                King of the Fingers (historical)
</p>,
            ],
            "className": "flex-data",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "Lord of Heart's Home",
              "nodeType": "host",
              "props": Object {
                "children": "Lord of Heart's Home",
                "className": "flex-me",
              },
              "ref": null,
              "rendered": "Lord of Heart's Home",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": "King of the Fingers (historical)",
              "nodeType": "host",
              "props": Object {
                "children": "King of the Fingers (historical)",
                "className": "flex-me",
              },
              "ref": null,
              "rendered": "King of the Fingers (historical)",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Coat of Arms",
          },
          "ref": null,
          "rendered": "Coat of Arms",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Three black ravens in flight...",
          },
          "ref": null,
          "rendered": "Three black ravens in flight...",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Ancestral Weapons",
          },
          "ref": null,
          "rendered": "Ancestral Weapons",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                Lady Forlorn
</p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "Lady Forlorn",
              "nodeType": "host",
              "props": Object {
                "children": "Lady Forlorn",
              },
              "ref": null,
              "rendered": "Lady Forlorn",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h3",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
